pandas/tests/reductions/test_reductions.py

```diff
@@ -880,16 +888,18 @@ def test_all_any_params(self):
         tm.assert_series_equal(s.all(level=0), Series([False, True, False]))
         tm.assert_series_equal(s.any(level=0), Series([False, True, True]))
 
-        # bool_only is not implemented with level option.
-        with pytest.raises(NotImplementedError):
+        msg = "Option bool_only is not implemented with option level"
+        with pytest.raises(NotImplementedError, match=msg):
             s.any(bool_only=True, level=0)
-        with pytest.raises(NotImplementedError):
+        with pytest.raises(NotImplementedError, match=msg):
             s.all(bool_only=True, level=0)
 
         # bool_only is not implemented alone.
-        with pytest.raises(NotImplementedError):
+        msg = "Series.any does not implement numeric_only"
+        with pytest.raises(NotImplementedError, match=msg):
             s.any(bool_only=True)
-        with pytest.raises(NotImplementedError):
+        msg = "Series.all does not implement numeric_only."
```

Comment:
This is a wrong error message, since it complains about numeric_only, but the actual code passes `bool_only`.

(so if we start asserting the message, let's take the opportunity to ensure it's a correct message)

Reply:
Ok - I made an attempt. Take a look at the changes to pandas/core/generic.py and please let me know if I handled it the right way. I wasn't quite sure which method was the right one to fix the error message.

Reply:
see below, this is too complicated to fix in this PR